File: Java/LeetCode/Left and Right Sum Differences.java
// https://leetcode.com/problems/left-and-right-sum-differences

class Solution {
    public int[] leftRightDifference(int[] nums) {
        int[] answer = new int[nums.length];

        for (int i = 0; i < nums.length; ++i) {
            int leftSum = 0, rightSum = 0;

            for (int j = 0; j < i; ++j) {
                leftSum += nums[j];
            }
            for (int j = nums.length-1; j > i; --j) {
                rightSum += nums[j];
            }

            answer[i] = Math.abs(leftSum - rightSum);
        }

        return answer;
    }
}

File: Kotlin/LeetCode/Left and Right Sum Differences.kt
// https://leetcode.com/problems/left-and-right-sum-differences

class Solution {
    fun leftRightDifference(nums: IntArray): IntArray {
        var answer = IntArray(nums.size);

        for (i in 0 until nums.size) {
            var leftSum = 0;
            var rightSum = 0;

            for (j in 0 until i) {
                leftSum += nums[j];
            }
            for (j in nums.size-1 downTo i+1) {
                rightSum += nums[j];
            }

            answer[i] = Math.abs(leftSum - rightSum);
        }

        return answer;
    }
}
